llvm/lib/IR/IRPrintingPasses.cpp

```diff
@@ -23,6 +23,8 @@
 
 using namespace llvm;
 
+extern cl::opt<bool> WriteNewDbgInfoFormat;
```

Comment:
This makes llvm/lib/IR depend on llvm/lib/IRPrinter (where this is defined), but the latter already depends on the former. That is, this introduces a cyclic dependency.

The definition of this symbol should probably live in this file and be extern in IRPrinter?

Reply:
Agree with the analysis. If you link the executable with [`ld.lld --warn-backrefs`](https://maskray.me/blog/2021-06-13-dependency-related-linker-options#warn-backrefs), you will get something like

```
error: backward reference detected: WriteNewDbgInfoFormat in llvm/_objs/Core/IRPrintingPasses.pic.o refers to llvm/_objs/IRPrinter/IRPrintingPasses.pic.o
```

(This is from a Bazel build).